modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/capabilities/CapabilitiesServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+
+package org.nuxeo.runtime.capabilities;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toMap;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.nuxeo.common.Environment.DISTRIBUTION_HOTFIX;
+import static org.nuxeo.common.Environment.DISTRIBUTION_NAME;
+import static org.nuxeo.common.Environment.DISTRIBUTION_SERVER;
+import static org.nuxeo.common.Environment.DISTRIBUTION_VERSION;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.model.ComponentContext;
+import org.nuxeo.runtime.model.ComponentManager;
+import org.nuxeo.runtime.model.ComponentStartOrders;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+/**
+ * @since 11.5
+ */
+public class CapabilitiesServiceImpl extends DefaultComponent implements CapabilitiesService {
+
+    public static final String CAPABILITY_SERVER = "server";
+
+    protected final Map<String, Supplier<Map<String, Object>>> capabilitiesSuppliers = new HashMap<>();
+
+    @Override
+    public void activate(ComponentContext context) {
+        super.activate(context);
+        new ComponentManager.Listener() {
+            @Override
+            public void beforeStart(ComponentManager mgr, boolean isResume) {
+                capabilitiesSuppliers.clear();
+            }
+        }.install();
```

Comment:
FYI this listener should be uninstalled too: in unit tests i see mutliple listener instances of this listener being piled up on the component manager (working on #4544)

Reply:
Thanks for the notice, could I wait for #4544 to be merged to fix it?

Reply:
Yes as #4544 will fix it (I migrated it so it will be uninstalled automatically), that was just as heads up in case this is backported on 10.10 or on the 2021 LTS branch (#4544 is not planned to be on the LTS for now)